packages/protocol/src/action-protocol/contexts.ts

```diff
@@ -14,45 +14,83 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 
-import { isArray, isObject, isString } from '../utils/typeguard-util';
-import { generateRequestId, isActionKind, RequestAction, ResponseAction } from './base-protocol';
+import { hasArrayProp, hasObjectProp, hasStringProp } from '../utils/type-util';
+import { Action, RequestAction, ResponseAction } from './base-protocol';
 import { Args, EditorContext, LabeledAction } from './types';
 
 /**
  * The `RequestContextActions` is sent from the client to the server to request the available actions for the context with id contextId.
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `RequestContextActions`.
  */
-export class RequestContextActions implements RequestAction<SetContextActions> {
-    static readonly KIND = 'requestContextActions';
-    constructor(
-        public readonly contextId: string,
-        public readonly editorContext: EditorContext,
-        public readonly requestId: string = generateRequestId(),
-        public readonly kind: string = RequestContextActions.KIND
-    ) {}
+export interface RequestContextActions extends RequestAction<SetContextActions> {
+    /**
+     * The unique action kind.
+     */
+    kind: typeof RequestContextActions.KIND;
+
+    /**
+     * The identifier for the context.
+     */
+    contextId: string;
+
+    /**
+     * The current editor context.
+     */
```

Comment:
I'd spare such comments, also if it hurts to be "incomplete". But imho they just add noise and add no information.

Reply:
I'm fine with omitting comments for self explaining primitive properties.
However, for object properties I think it would be beneficial to add a comment which links to the type of the property.
This way a user can just ctrl+ click to the definition if something is not clear.